x/evm/watcher/querier.go

```diff
@@ -181,3 +199,61 @@ func (q Querier) getTransactionByBlockAndIndex(block *EthBlock, idx uint) (*rpct
 	}
 	return nil, errors.New("no such transaction in target block")
 }
+
+func (q Querier) GetAccount(addr sdk.AccAddress) (*types.EthAccount, error) {
+	if !q.enabled() {
+		return nil, errors.New(MsgFunctionDisable)
+	}
+	var acc types.EthAccount
+	b, e := q.store.Get([]byte(GetMsgAccountKey(addr.String())))
+	if e != nil {
+		return nil, e
+	}
+	e = json.Unmarshal(b, &acc)
+	if e != nil {
+		return nil, e
+	}
+	return &acc, nil
+}
+
+func (q Querier) GetState(addr common.Address, key []byte) ([]byte, error) {
+	if !q.enabled() {
+		return nil, errors.New(MsgFunctionDisable)
+	}
+	b, e := q.store.Get([]byte(GetMsgStateKey(addr, key)))
+	if e != nil {
+		return nil, e
+	}
+	ret := hexutils.HexToBytes(string(b))
+	return ret, nil
+}
+
+func (q Querier) GetParams() (*evmtypes.Params, error) {
+	if !q.enabled() {
+		return nil, errors.New(MsgFunctionDisable)
+	}
+	b, e := q.store.Get([]byte(prefixParams))
+	if e != nil {
+		return nil, e
+	}
+	var params evmtypes.Params
+	e = json.Unmarshal(b, &params)
+	if e != nil {
+		return nil, e
+	}
+	return &params, nil
+}
+
+func (q Querier) HasContractBlockedList(key []byte) bool {
+	if !q.enabled() {
+		return false
+	}
+	return q.store.Has([]byte(prefixBlackList + hexutils.BytesToHex(key)))
```

Comment:
It's better for '[]byte appending' instead of 'string +' with consideration of the efficiency.